tasks/Task.cpp

```diff
@@ -1,39 +1,103 @@
 /* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
 
 #include "Task.hpp"
+#include <base-logging/Logging.hpp>
 
 using namespace webrtc_rustysignal;
+using namespace std;
 
 Task::Task(std::string const& name)
     : TaskBase(name)
 {
+    Json::CharReaderBuilder builder;
+    m_json_reader = builder.newCharReader();
+
+    _connection_timeout.set(base::Time::fromSeconds(1));
+    _close_timeout.set(base::Time::fromSeconds(1));
 }
 
 Task::~Task()
 {
+    delete m_json_reader;
 }
 
-
-
-/// The following lines are template definitions for the various state machine
-// hooks defined by Orocos::RTT. See Task.hpp for more detailed
-// documentation about them.
-
 bool Task::configureHook()
 {
-    if (! TaskBase::configureHook())
+    if (!TaskBase::configureHook())
         return false;
+
+    promise<void> promise;
+    auto future = promise.get_future();
+    m_ws.onOpen([&promise]() {
+        LOG_DEBUG_S << "websocket opened";
+        promise.set_value();
+    });
+    m_ws.onError([&promise](string const& error) {
+        LOG_DEBUG_S << "websocket error" << error;
+        promise.set_exception(make_exception_ptr(runtime_error(error)));
+    });
+
+    m_ws.open(_url.get() + "?user=" + _peer_id.get());
```

Comment:
Just a question: is the websocket always opened by passing the peer_id as a user parameter?

Reply:
Not sure what you mean by that. It is what rustysignal expects. Websockets in general are only a communication channel.